### ISSUE TITLE
Major preformance drop v0.5.16

### ISSUE THREAD

**Issue by nukeline** [processing/p5.js#2465]
Updated from v0.5.8 to v0.5.16 and now I get 70% frame drop. I'm not using WEBGL, just drawing a bunch of images etc.

(firefox/chrome desktop win10)

**Comment by nukeline** [comment]
Checked the preformance in chrome, on v0.5.8 drawImage() took 58% of the total time (highest). On v0.5.16 "(anonymous)" took 98.6% of total time (drawImage second place at 22.3%). 

(anonymous) is referenced to line p5.js:44898. 

I have no idea what anonymous does but it's seriously screwing things up for me

**Comment by nukeline** [comment]
I found the bug.

`
stroke(255);
//ellipse(x, y, etc);
`

I forgot to comment stroke when i commented ellipse. The code does not handle the single stroke() call very well in this version.

**Comment by mlarghydracept** [comment]
Thanks for submitting the issue! In the future try to give more of a code sample for recreating the issue. It is more helpful for us. Based on your last comment though I just decided to write something simple to test with:

```
function setup() {
    createCanvas(windowWidth,windowHeight);
}

function draw() {
	for(var i = 0; i<400; i++) {
		stroke(0);
		ellipse(random(0, width), random(0,height), 100, 100);
	}
}
``` 
And I was able to recreate your problem. This seems to be due to `validateParameters()` in the friendly-error-system. If you switch to the minified (p5.min.js) version of the library the slowdown is not present even in 0.5.16

See the [last two comments in this issue](https://github.com/processing/p5.js/issues/2190)
